File: src/main/java/io/github/jmkeyes/spring/boot/thrift/server/ThriftController.java
package io.github.jmkeyes.spring.boot.thrift.server;

import org.apache.thrift.protocol.TJSONProtocol;
import org.apache.thrift.protocol.TProtocolFactory;

import java.lang.annotation.*;

/**
 * An annotation to mark the class as a Thrift service controller.
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
public @interface ThriftController {
    /**
     * At least one URL mapping under which this Thrift service should be available.
     *
     * @return At least one {@link String} path to mount the servlet at.
     */
    String[] value() default {};

    /**
     * The protocol factory this Thrift service should communicate with.
     *
     * @return The {@link TProtocolFactory} this Thrift service should use.
     */
    Class<? extends TProtocolFactory> protocolFactory() default TJSONProtocol.Factory.class;
}

File: src/main/java/io/github/jmkeyes/spring/boot/thrift/server/ThriftController.kt
package io.github.jmkeyes.spring.boot.thrift.server

import org.apache.thrift.protocol.TJSONProtocol
import org.apache.thrift.protocol.TProtocolFactory
import java.lang.annotation.Inherited
import kotlin.reflect.KClass

@Inherited
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class ThriftController (
        /**
         * At least one URL mapping under which this Thrift service should be available.
         */
        vararg val value: String = [],

        /**
         * The protocol factory this Thrift service should communicate with.
         */
        val protocolFactory: KClass<out TProtocolFactory> = TJSONProtocol.Factory::class
)
